### PR TITLE
Reduce caching of unused directories in CircleCI

### PR DESCRIPTION
**Pull request by stuarthicks**

### CODE REVIEW

**Comment by stuarthicks** [comment]
Circle will pass when someone with access does a "rebuild without cache".

File: circle.yml [hunk 1]
```diff
@@ -1,26 +1,40 @@
 machine:
   environment:
-    AWS_ACCESS_KEY_ID: a
-    AWS_SECRET_ACCESS_KEY: b
+    AWS_ACCESS_KEY_ID: 1
+    AWS_SECRET_ACCESS_KEY: 1
+    GO15VENDOREXPERIMENT: 1
+    GOPATH: $HOME/gocode
+    PATH: $GOPATH/bin:$PATH
+    ZENPATH: $HOME/gocode/src/github.com/zencoder
```
Comment (brycefisher) [inline circle.yml]: Aw, no `FABPATH`? :sob:

File: circle.yml [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
   override:
-    - go get github.com/tools/godep
-    - go get golang.org/x/tools/cmd/cover
-    - go get github.com/mattn/goveralls
-    - go get github.com/modocache/gover
+    - type cover     || go get golang.org/x/tools/cmd/cover
+    - type goveralls || go get github.com/mattn/goveralls
+    - type gover     || go get github.com/modocache/gover
+  post:
+    - mv ~/.go_workspace /tmp || echo
```
Comment (brycefisher) [inline circle.yml:30]: No more `rm`ing these unused dirs? Why the `|| echo`? Seems like a simple `;` would do it, but nbd either way.

Comment (stuarthicks) [inline circle.yml:30]: Well the `mv` is faster. Some of these dirs come pre-populated, so this saves a few more seconds. However, sometimes these dirs don't exist, so the `mv` fails, hence the echo.

**Comment by thomshutt** [comment]
`&& make coveralls` <-- We don't have a coveralls job for this project

**Comment by stuarthicks** [comment]
Interesting, given we do have the Makefile goal. We could create a coveralls job for it...

**Pull request closed without merging**